src/components/oscal-utils/OSCALPropUtils.js

```diff
@@ -0,0 +1,33 @@
+/**
+ * When a namespace is not specified for a property, the assumption is that the
+ * namespace is default namespace.
+ */
+const NIST_DEFAULT_NAMESPACE = "https://csrc.nist.gov/ns/oscal";
+
+/**
+ * Return the given namespace, or if undefined, the default namespace.
+ */
+export function namespaceOf(ns) {
+  return ns ?? NIST_DEFAULT_NAMESPACE;
+}
+
+/**
+ * Returns the first matching property in the given list.
+ *
+ * @param props the list of properties to check
+ * @param filter the attributes to match against
+ */
+export default function matchingProp(props, filter) {
```

Comment:
```suggestion
/**
 * @callback PropertyFilterValueMatcher
 * @param {string} value - the value to check against
 * @returns {boolean} whether the given property matches
 */

/**
 * A filter to apply to properties to find a matching property.
 *
 * This allows matching based on the namespace (defaulting to the default namespace),
 * the name, and the value. Exactly one of `value` or `filter` must be supplied.
 *
 * @typedef PropertyFilter
 * @type {object}
 * @property {string|undefined} ns - the namespace to check in
 * @property {string| name - the name of the field to match
 * @property {string} value - the value of the field to match
 * @property {PropertyFilterValueMatcher} filter - a function to match the value against
 */

/**
 * Returns the first matching property in the given list.
 *
 * @param props {Array.<Object>} the list of properties to check
 * @param filter {PropertyFilter} the attributes to match against
 */
export default function matchingProp(props, filter) {
```

I think we can represent the types from before as JSDoc types if we want to... I don't know whether this is useful enough or not.

Reply:
Would this info make more sense to add once we add TypeScript? . . . or maybe this could be helpful to who implements TS here?

Reply:
Fair yeah -- maybe we just add it then.

Reply:
Reviewing over it again, this documentation is valuable even now with JS and TS support is just around the corner.